protostar/configuration_file/configuration_file.py

```diff
@@ -32,6 +46,16 @@ def get_command_argument(
     ]:
         ...
 
+    @abstractmethod
+    def create_model(
+        self,
+    ) -> TConfigurationFileModel:
+        ...
+
+    @abstractmethod
+    def save(self, configuration_file_model: TConfigurationFileModel) -> Path:
```

Comment:
why is this returning file path?

Reply:
It was convenient from the test perspective.

Reply:
Aren't tests able to get this value from elsewhere? It doesn't really make any sense to me to have such contract here.

Reply:
I think it makes sense because`ConfigurationFile`'s callers don't know the format of the file, hence they can't know the exact path upfront.

Reply:
hmm but now they cannot learn about the file without saving it...

would it make sense to expose the path as a `property` here?

Reply:
I removed the `save` method from the `ConfigurationFile` because readers don't care about the file format, but writers do.